metric_map.go

```diff
@@ -176,6 +177,80 @@ func (mm *MetricMap) Split(count int) []*MetricMap {
 	return maps
 }
 
+func tagsMatch(tagNames []string, tagsKey string) string {
+	res := make([]string, 0)
+	for _, tv := range strings.Split(tagsKey, ",") {
+		for _, tagName := range tagNames {
+			if strings.HasPrefix(tv, tagName+":") {
+				res = append(res, tv)
+				break
+			}
+		}
+	}
+	return strings.Join(res, ",")
+}
+
+func (mm *MetricMap) SplitByTags(tagNames []string) map[string]*MetricMap {
+	maps := make(map[string]*MetricMap)
```

Comment:
Is this going to be okay if each of these `Each` functions execute async?

Reply:
`Each` isn't async